src/base.jl

```diff
@@ -437,6 +437,23 @@ function add_service!(sys::System, service::Service, contributing_devices; kwarg
     end
 end
 
+"""
+Similar to [`add_component!`](@ref) but for StaticReserveGroup.
+
+# Arguments
+- `sys::System`: system
+- `service::StaticReserveGroup`: service to add
+"""
+function add_service!(sys::System, service::StaticReserveGroup; kwargs...)
+    if sys.runchecks && !validate_struct(sys, service)
+        throw(InvalidValue("Invalid value for $service"))
+    end
+
```

Comment:
From reading your test it looks like the user workflow is to

1. Add services to the system.
2. Add service group to the system.

That being the case, I think that we should verify that the individual services are already part of the system.  Otherwise, someone could get the system into a strange state.

Would you mind doing this?  It should be something like
```
for _service in get_contributing_services(service)
    if !is_attached(_service, sys)
        throw(ArgumentError("service $(get_name(_service)) is not part of the system"))
    end
end

That raises another question.  What would you like to happen if a user calls `remove_component!(sys, individual_service)` when that service is part of a group?  My inclination is that we should block that and force the user to remove the group before removing a service.  I'll offer to help covering these corner cases.

Reply:
This is a great point! After the refactoring, I will add this!

> That raises another question.  What would you like to happen if a user calls `remove_component!(sys, individual_service)` when that service is part of a group?  My inclination is that we should block that and force the user to remove the group before removing a service.  I'll offer to help covering these corner cases.

Indeed, I will come back to this and talk it over with you, as this will be the more difficult part of the change.

Reply:
done!